File: core/envelope-sim/src/testFixtures/java/fr/sncf/osrd/envelope_sim/FlatPath.java
package fr.sncf.osrd.envelope_sim;

import fr.sncf.osrd.path.interfaces.PhysicsPath;

public class FlatPath implements PhysicsPath {
    private final double length;
    private final double slope;

    public FlatPath(double length, double slope) {
        this.length = length;
        this.slope = slope;
    }

    @Override
    public double getLength() {
        return length;
    }

    @Override
    public double getAverageGrade(double begin, double end) {
        return slope;
    }

    @Override
    public double getMinGrade(double begin, double end) {
        return slope;
    }
}

File: core/envelope-sim/src/testFixtures/java/fr/sncf/osrd/envelope_sim/FlatPath.kt
package fr.sncf.osrd.envelope_sim

import fr.sncf.osrd.path.interfaces.Electrification
import fr.sncf.osrd.path.interfaces.PhysicsPath
import fr.sncf.osrd.utils.DistanceRangeMap

class FlatPath(override val length: Double, private val slope: Double) : PhysicsPath {
    override fun getAverageGrade(begin: Double, end: Double): Double {
        return slope
    }

    override fun getMinGrade(begin: Double, end: Double): Double {
        return slope
    }

    override fun getElectrificationMap(
        basePowerClass: String?,
        powerRestrictionMap: DistanceRangeMap<String>?,
        powerRestrictionToPowerClass: Map<String, String>?,
        ignoreElectricalProfiles: Boolean,
    ): DistanceRangeMap<Electrification> {
        TODO("Not yet implemented")
    }
}
